File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.java
import java.io.*;
import java.util.HashMap;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        HashMap<Integer, Integer> cardsdict = new HashMap<>();

        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            int num = Integer.parseInt(st.nextToken());
            if (cardsdict.containsKey(num)) {
                cardsdict.put(num, cardsdict.get(num) + 1);
            } else {
                cardsdict.put(num, 1);
            }
        }

        int M = Integer.parseInt(br.readLine());

        st = new StringTokenizer(br.readLine());
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < M; i++) {
            int query = Integer.parseInt(st.nextToken());
            if (cardsdict.containsKey(query)) {
                result.append(cardsdict.get(query)).append(" ");
            } else {
                result.append(0).append(" ");
            }
        }

        System.out.println(result.toString().trim());

        br.close();
    }
}

File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.py
import sys

N = int(sys.stdin.readline())
cards = list(map(int, sys.stdin.readline().split()))
M = int(sys.stdin.readline())
checks = list(map(int, sys.stdin.readline().split()))

cardsdict = {}
for num in cards:
    if num in cardsdict:
        cardsdict[num] += 1
    else:
        cardsdict[num] = 1

for check in checks:
    if check in cardsdict:
        print(cardsdict[check], end=" ")
    else:
        print(0, end=" ")